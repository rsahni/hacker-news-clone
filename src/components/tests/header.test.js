import React from 'react';
import { render, queryByTestId } from '@testing-library/react';
import '@testing-library/jest-dom'
import Header from '../header';

test('renders header', () => {
    render(<Header />);
    expect(
        queryByTestId(document.documentElement, 'header'),
    ).toBeInTheDocument()
});

test('renders header matches snapshot', () => {   
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
});