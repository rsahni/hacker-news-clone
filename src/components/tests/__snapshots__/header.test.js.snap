// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header matches snapshot 1`] = `
<DocumentFragment>
  <header
    data-testid="header"
  >
    <div
      class="hn-header-section"
    >
      <div
        class="site-title-mobile"
      >
        <h1>
          HN Clone Rahul
        </h1>
      </div>
      <div
        class="hn-header-item-list"
      >
        <div
          class="hn-header-item first-column"
        >
          Comments
        </div>
        <div
          class="hn-header-item second-column"
        >
          View Count
        </div>
        <div
          class="hn-header-item third-column"
        >
          UpVote
        </div>
        <div
          class="hn-header-item fourth-column"
        >
          News Details
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
